app/controllers/gems_adoptions_controller.rb

```diff
@@ -20,6 +20,15 @@ def create
     end
   end
 
+  def update
+    adoption_request = AdoptionRequest.find(params[:id])
+    gems_adoption = GemsAdoption.find(adoption_request.gems_adoption_id)
+    gems_adoption.status = 'close'
+    gems_adoption.save
+    gem_ownership_transfer = GemOwnershipTransfer.create(old_user_id: current_user.id, new_user_id: adoption_request.user_id, ruby_gem_id: gems_adoption.ruby_gem_id)
+    redirect_to gem_path(gems_adoption.ruby_gem)
+  end
```

Comment:
We need to be consider the case where the update of `status` of the `GemAdoption` fails or the `GemOwnershipTransfer` can't be created. What do you think?